sfcasts/crud-controller.md

```diff
@@ -1,171 +1,90 @@
+
+Okay, go refresh to reveal our new link, click and... whoa! This is super cool! We have an `isApproved` slider, which saves automatically. We also have a search bar on top... and sortable columns to help us find whatever we're looking for.
+
+We can edit, delete the edit, and it even has a cool calendar. We are absolutely loaded with awesome features. It's pretty cool! So let's repeat this for our other three controllers.
+
+I'll head back to my terminal and run:
+
+```terminal
+Symfony console make:admin:crud
+```
+
+Once again, let's get one for `Answer` with the defaults stuff... And one for `Topic` with the default stuff. I'll clear my screen... And the last one we need is for `User`.
+
+Beautiful! The only thing it did is three more of these CRUD controller classes, but to make those useful, we need to link to them. So, I'm going to paste three more links here. Let's say `Answers`... And I'll customize each of these links.
+
+Nice! I'll refresh... and look! We now have four different admin sections. All right. I want to look just a little deeper into how this is working behind the scenes. Go to `QuestionCrud` controller and look at its base class. I'll hold "cmd + ctrl" and jump into `AbstractCrudController`. We saw earlier that our dashboard extends an `AbstractDashboardController`. Our CRUD controllers extend an `AbstractCrudController`.
+
+Pretty much everything about how our CRUD controller works is going to be controlled by overriding these configure methods that you see inside of here. We'll learn about all of these as we go along, but on a high level, `configureCrud` is here to help you configure the entire CRUD section. `configureAssets` allows you to control custom CSS and JavaScript in this section, `configureActions` allows you to control the actions you want, which are things like having an index page, having an edit action, or having a delete action. More on that later.
+
+The last important one is `configureFields`, which is going to configure which fields we see on both the index page and the form. But don't worry about those yet. We'll see all of them later.
+
+Then, below this, you can actually see all of the code that executes the pages. So `index` is our index page, `detail` is a detail show page, `edit` is the edit form.
+
+You can see the code that actually runs behind all of this, which is super useful when we're figuring out how to extend things. But... wait a second. If you scroll back up here to the configure methods, a few of these look familiar. Some of them *also* exist in the dashboard based controller class, specifically `configureAssets`, `configureCrud`, `configureActions`, and `configureFilters`. These all live side of `AbstractDashboardController`. There's `configureAssets`, and then further down here, CRUD actions and filters.
```

Comment:
Is "dashboard based controller class" correct? I had no idea.

Reply:
That's supposed to be "These all live "inside" of..."
I changed it in my draft.